godel_simple_gui/include/godel_simple_gui/options/surface_detection_configuration.h

```diff
@@ -0,0 +1,36 @@
+#ifndef SURFACE_DETECTION_CONFIGURATION_H
+#define SURFACE_DETECTION_CONFIGURATION_H
+
+#include <godel_simple_gui/parameter_window_base.h>
+#include <ui_surface_detection_configuration.h>
+#include <godel_msgs/SurfaceDetectionParameters.h>
+
+namespace Ui {
+  class SurfaceDetectionConfigWindow;
+}
+
+namespace godel_simple_gui
+{
+
+class SurfaceDetectionConfigWidget : public ParameterWindowBase
+{
+  Q_OBJECT
+public:
+  SurfaceDetectionConfigWidget(godel_msgs::SurfaceDetectionParameters params);
+
+  godel_msgs::SurfaceDetectionParameters& params() { return params_; }
+  const godel_msgs::SurfaceDetectionParameters& params() const { return params_; }
+
+  virtual void update_display_fields();
+
+  virtual void update_internal_fields();
```

Comment:
There are several methods in this PR that are named with lowercase and underscores.  As this is the case in many other packages in this repo we should address this in a separate PR as it appears to require a rather large set of changes.